### ISSUE TITLE
Code generated by the `--target` option of the `new` command should have use declarations

### ISSUE THREAD

**Issue by peterhuene** [bytecodealliance/cargo-component#89]
Currently, the source code generated by the `--target` option of the `new` command fully qualifies every type name when generating the trait implementations of any exports for the target world.

Instead, it should be smarter and generate `use` declarations of the referenced types, falling back to a qualified path only when there would be a conflict.

**Comment by peterhuene** [comment]
I have this implemented, but need #88 merged first.

**Comment by peterhuene** [comment]
Example output from a managed `wasi:http/proxy` target (introduced the `custom` and `custom2` exports to show a conflicting name):

```rust
use bindings::exports::wasi::http::{
    custom::{Custom, InputStream, OutputStream},
    custom2::{Custom2, Pollable},
    incoming_handler::{IncomingHandler, IncomingRequest, ResponseOutparam},
};

struct Component;

impl IncomingHandler for Component {
    fn handle(request: IncomingRequest, response_out: ResponseOutparam) {
        unimplemented!()
    }
}

impl Custom for Component {
    fn foo(input: InputStream) -> OutputStream {
        unimplemented!()
    }
}

impl Custom2 for Component {
    fn foo(p: Pollable) -> bindings::exports::wasi::http::custom2::OutputStream {
        unimplemented!()
    }
}

bindings::export!(Component);
```
